aptos-move/framework/aptos-framework/sources/vesting.move

```diff
@@ -0,0 +1,451 @@
+    public entry fun distribute(admin: &signer, pool_address: address) acquires VestingPool {
+        let vesting_pool = borrow_global<VestingPool>(pool_address);
+        verify_admin(admin, vesting_pool);
+        assert_active_vesting_pool(vesting_pool);
+
+        let (_, total_withdrawable, _, _) = stake::get_stake(pool_address);
+        let coins_to_distribute =
+            stake::withdraw_with_cap(&vesting_pool.stake_pool_owner_cap, total_withdrawable);
+        let distribute_amount = (coin::value(&coins_to_distribute) as u128);
+
+        let len = vector::length(&vesting_pool.shareholders);
+        let i = 0;
+        let total_shares = (vesting_pool.grant_amount as u128);
+        while (i < len) {
+            let shareholder = *vector::borrow(&vesting_pool.shareholders, i);
+            let shares = (*simple_map::borrow(&vesting_pool.shares, &shareholder) as u128);
+            // u128 math to avoid overflow from multiplication. Then convert back to u64 after division.
+            // Multiplication first to minimize rounding error.
+            let amount = ((distribute_amount * shares / total_shares) as u64);
+            let coins = coin::extract(&mut coins_to_distribute, amount);
+            coin::deposit(shareholder, coins);
+
+            i = i + 1;
+        };
+
+        // Send any remaining "dust" (leftover due to rounding error) to the withdrawal address.
+        coin::deposit(vesting_pool.withdrawal_address, coins_to_distribute);
+    }
+
+    public entry fun pause_vesting_pool(admin: &signer, pool_address: address) acquires VestingPool {
+        let vesting_pool = borrow_global_mut<VestingPool>(pool_address);
+        verify_admin(admin, vesting_pool);
+        assert_active_vesting_pool(vesting_pool);
+
+        vesting_pool.state = VESTING_POOL_PAUSED;
+    }
+
+    public entry fun unpause_vesting_pool(admin: &signer, pool_address: address) acquires VestingPool {
+        let vesting_pool = borrow_global_mut<VestingPool>(pool_address);
+        verify_admin(admin, vesting_pool);
+        assert!(vesting_pool.state == VESTING_POOL_PAUSED, error::invalid_state(EVESTING_POOL_NOT_PAUSED));
+
+        vesting_pool.state = VESTING_POOL_ACTIVE;
+    }
+
+    public entry fun terminate_vesting_pool(admin: &signer, pool_address: address) acquires VestingPool {
+        let vesting_pool = borrow_global_mut<VestingPool>(pool_address);
+        verify_admin(admin, vesting_pool);
+        assert_active_vesting_pool(vesting_pool);
+
+        // Cannot terminate the vesting pool if there's an in-flight distribution.
+        let (_, withdrawable, _, _pending_unlocked) = stake::get_stake(pool_address);
+        assert!(withdrawable + _pending_unlocked == 0, error::invalid_state(EDISTRIBUTIONS_IN_FLIGHT));
+
+        vesting_pool.state = VESTING_POOL_TERMINATED;
+
+        // Request unlock of all stake.
+        let (active, _, pending_active, _) = stake::get_stake(pool_address);
```

Comment:
forgive me, but can we not move `pending_active` back?

Reply:
Unfortunately pending_active cannot currently be moved directly to inactive.